DockerUtil.go

```diff
@@ -37,13 +33,56 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
 )
 
-func StartDockerDaemon() {
-	dockerdStart := "dockerd --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
-	out, _ := exec.Command("/bin/sh", "-c", dockerdStart).Output()
-	log.Println(string(out))
-	waitForDockerDaemon(retryCount)
+func StartDockerDaemon(dockerConnection, dockerRegistryUrl, dockerCert string) {
+	connection := dockerConnection
+	fmt.Println(connection)
+	u, err := url.Parse(dockerRegistryUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if connection == "secure-with-cert" {
+		os.MkdirAll("/etc/docker/certs.d/"+u.Host, os.ModePerm)
+		f, err := os.Create("/etc/docker/certs.d/" + u.Host + "/ca.crt")
+
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		defer f.Close()
+
+		_, err2 := f.WriteString(dockerCert)
+
+		if err2 != nil {
+			log.Fatal(err2)
+		}
+		dockerdStart := "dockerd --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
+		out, _ := exec.Command("/bin/sh", "-c", dockerdStart).Output()
+		logStage("Secure with Cert")
+
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
+	} else if connection == "insecure" {
+		dockerdstart := "dockerd --insecure-registry " + dockerRegistryUrl + " --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
+		out, _ := exec.Command("/bin/sh", "-c", dockerdstart).Output()
+		logStage("Insecure Registry")
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
+	} else {
```

Comment:
can we remove this step and put this inside secure block

Reply:
Done Sir